environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -175,6 +201,38 @@ private EnvironmentDto changeCredential(String accountId, String environmentName
         return environmentDtoConverter.environmentToDto(saved);
     }
 
+    private EnvironmentDto updateAwsDiskEncryptionParameters(String accountId, String environmentNameOrCrn, AwsDiskEncryptionParametersDto dto,
+            Environment environment) {
+            if (dto.getEncryptionKeyArn() != null) {
+                ValidationResult validateKey = environmentService.getValidatorService().validateEncryptionKeyArn(dto.getEncryptionKeyArn(),
+                        accountId);
+                if (!validateKey.hasError()) {
+                    if (environment.getParameters() == null) {
+                        AwsParameters awsParameters = new AwsParameters();
+                        awsParameters.setEncryptionKeyArn(dto.getEncryptionKeyArn());
+                        awsParameters.setAccountId(accountId);
+                        awsParameters.setId(environment.getId());
+                        awsParameters.setName(environment.getName());
+                        awsParameters.setEnvironment(environmentViewConverter.convert(environment));
+                        environment.setParameters(awsParameters);
+                    } else {
+                        AwsParameters awsParameters = (AwsParameters) environment.getParameters();
+                        if (awsParameters.getEncryptionKeyArn() == null) {
+                            ((AwsParameters) (environment.getParameters())).setEncryptionKeyArn(dto.getEncryptionKeyArn());
```

Comment:
`(AwsParameters) (environment.getParameters())` is already available as `awsParameters`, so this can be simplified.